src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1193,7 +1193,9 @@ private BoundExpression SynthesizeMethodGroupReceiver(CSharpSyntaxNode syntax, A
             HashSet<DiagnosticInfo> unused = null;
             if (currentType.IsEqualToOrDerivedFrom(declaringType, ignoreDynamic: false, useSiteDiagnostics: ref unused))
             {
-                return ThisReference(syntax, currentType, wasCompilerGenerated: true);
+                // PROTOTYPE: Get the type of `this` in a better way. Probably want to modify the above if statement?
+                var typeOfThis = currentType.ExtensionClassType ?? currentType;
+                return ThisReference(syntax, typeOfThis, wasCompilerGenerated: true);
```

Comment:
It took me a bit to get around to it, but I'll be responding to your comments that are not addressed in the new commits I just pushed. A re-review once I'm done would be nice.

I have changed this code a bit while responding to your feedback, but there's a bit of complex behavior here that needs to be resolved (it is a bigger scope than this PR). Specifically, binding of simple names in an extension class: what the lookup order is, how to resolve ambiguity, and what the type of things are.

The comment about the if statement in the original PR was a copy-paste error :frowning: (the actual use of `IsEqualToOrDerivedFrom` in the existing code is to check if the invoked member is actually part of the current class, and if it's not, bind it as an interactive receiver instead. This logic shouldn't change with the introduction of extension everything.)